### PR TITLE
Support admin registration with no internet

### LINKED ISSUE #967: Issue with local email authentication on self hosted instance of typebot
Hello.

I set up a local typebot instance on docker. 
Containers are starting and are accessible.
I am using traefik as reverse proxy. 

After typing my mail-address on the signup screen, nothing happens, no error message, no timeout, nothing.

image

In the typebot-builder container there is no logmessage as well.
In the Javascript console no messages as well.

The request to https://typebot.subdomain.domain.de/api/auth/signin/email? never comes back.

Post-Data from the request:
```
email: a@b.com
redirect: false
csrfToken: 8c4e8fd5d6a8f51014364a278306e7b302c2a228ebecc0e3e48a7ddedad1851c
callbackUrl: https://typebot.subdomain.domain.de/de/signin
json: true
```

My local mailserver is accessible without authentication - could that be a problem?

docker-compose.yml

```
version: '3.3'

volumes:
  db-data:

services:
  typebot-db:
    image: postgres:14-alpine
    restart: always
    volumes:
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=typebot
      - POSTGRES_PASSWORD=typebot
    networks:
      - typebot

  typebot-builder:
    image: baptistearno/typebot-builder:latest
    restart: always
    depends_on:
      - typebot-db
    env_file: .env
    labels:
      - traefik.enable=true
      - traefik.http.middlewares.typebot-redirect-https.redirectscheme.scheme=https
      - traefik.http.routers.typebot.middlewares=typebot-redirect-https
      - traefik.http.routers.typebot.rule=Host(`typebot.subdomain.domain.de`)
      - traefik.http.routers.typebot.entrypoints=http
      - traefik.http.routers.typebot-https.rule=Host(`typebot.subdomain.domain.de`)
      - traefik.http.routers.typebot-https.tls=true
      - traefik.http.routers.typebot-https.entrypoints=https
      - traefik.docker.network=traefik
      - traefik.http.services.typebot.loadbalancer.server.port=3000
    networks:
      - traefik
      - typebot

  typebot-viewer:
    image: baptistearno/typebot-viewer:latest
    restart: always
    env_file: .env
    labels:
      - traefik.enable=true
      - traefik.http.middlewares.bot-redirect-https.redirectscheme.scheme=https
      - traefik.http.routers.bot.middlewares=bot-redirect-https
      - traefik.http.routers.bot.rule=Host(`bot.subdomain.domain.de`)
      - traefik.http.routers.bot.entrypoints=http
      - traefik.http.routers.bot-https.rule=Host(`bot.subdomain.domain.de`)
      - traefik.http.routers.bot-https.tls=true
      - traefik.http.routers.bot-https.entrypoints=https
      - traefik.docker.network=traefik
      - traefik.http.services.bot.loadbalancer.server.port=3000
    networks:
      - traefik
      - typebot


networks:
  traefik:
    external: true
  typebot:
    external: true
```

.env
```
# Make sure to change this to your own random string of 32 characters (https://docs.typebot.io/self-hosting/docker#2-add-the-required-configuration)
ENCRYPTION_SECRET=YY2rXphT6zxtwa9+o5X822ojrdSr2CSz

DATABASE_URL=postgresql://postgres:typebot@typebot-db:5432/typebot

NEXTAUTH_URL=https://typebot.subdomain.domain.de
NEXT_PUBLIC_VIEWER_URL=https://bot.subdomain.domain.de

ADMIN_EMAIL=a@example.com
# For more configuration options check out: https://docs.typebot.io/self-hosting/configurationi

NEXTAUTH_URL_INTERNAL=http://typebot-builder:3000

SMTP_HOST=192.168.156.150
SMTP_PORT=25
SMTP_SECURE=false
SMTP_AUTH_DISABLED=false

NEXT_PUBLIC_SMTP_FROM=typebot@subdomain.domain.de

DEBUG=true
DISABLE_SIGNUP=false
DEFAULT_WORKSPACE_PLAN=UNLIMITED
```

Log messages in builder-container:
```
- event [next-runtime-env] read environment variables prefixed with 'NEXT_PUBLIC_' from process.env.
- ready [next-runtime-env] wrote browser runtime environment variables to '/app/apps/builder/public/__ENV.js'.
Waiting for 15s for database to be ready...
Prisma schema loaded from packages/prisma/postgresql/schema.prisma
Datasource "db": PostgreSQL database "typebot", schema "public" at "typebot-db:5432"

57 migrations found in prisma/migrations


No pending migrations to apply.
  ▲ Next.js 13.5.4
  - Local:        http://localhost:3000
  - Network:      http://0.0.0.0:3000

 ✓ Ready in 651ms
```

Docker Image is:
baptistearno/typebot-builder     latest      699d631d0f8e

Thank you.
Bye
Christoph

### PR DESCRIPTION
**Pull request by thisago**
I ran Typebot locally with Docker, and as expected, everything worked fine.

But I realized that offline user registering wasn't working. Then I found the issue #967, that reports the same problem. @Streethawk27 even reported that [it tries to connect on Github](https://github.com/baptisteArno/typebot.io/issues/967#issuecomment-1782675853).

At now, the error returned by backend is just "Typerror: fetch error", but the problem is the same.

What happens is that backend checks if the email provider isn't an disposable one, but it requests the disposable email list every time, making impossible to create an user account while offline.

On this small commit I just skipped this verification if email is same as `ADMIN_EMAIL` defined on `.env`

### CODE REVIEW

File: apps/builder/src/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -179,7 +179,7 @@ export const getAuthOptions = ({
       if (restricted === 'rate-limited') throw new Error('rate-limited')
       if (!account) return false
       const isNewUser = !('createdAt' in user && isDefined(user.createdAt))
-      if (isNewUser && user.email) {
+      if (isNewUser && user.email && user?.email != env.ADMIN_EMAIL) {
```
Comment (baptisteArno) [inline apps/builder/src/pages/api/auth/[...nextauth].ts:182]:
```suggestion
      if (isNewUser && user.email && user.email != env.ADMIN_EMAIL) {
```

**Review by baptisteArno — CHANGES_REQUESTED** [review]

**Comment by thisago** [comment]
@baptisteArno ping!

File: apps/builder/src/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ export const getAuthOptions = ({
       if (restricted === 'rate-limited') throw new Error('rate-limited')
       if (!account) return false
       const isNewUser = !('createdAt' in user && isDefined(user.createdAt))
-      if (isNewUser && user.email) {
+      if (isNewUser && user.email && user.email !== env.ADMIN_EMAIL) {
```
[the remark this answers is not in the document]
Comment (thisago) [reply apps/builder/src/pages/api/auth/[...nextauth].ts:182]: Oh, that's right, thanks for the fix

**Comment by ZimaBlueee** [comment]
@thisago 

图片

Hello, I have configured admin_email, but there is no input box on the page

My .env：
```
ENCRYPTION_SECRET=do+Us357Bxxxxxx
DATABASE_URL=postgresql://postgres:xxx@typebot-db:5432/typebot
NODE_OPTIONS=--no-node-snapshot
NEXTAUTH_URL=http://10.8.2.70:7017/
NEXT_PUBLIC_VIEWER_URL=http://10.8.2.70:7018/
ADMIN_EMAIL=xx@xxx.com
```

Then, I configured several non-existent SMTP configurations, and the input box appeared, but I still couldn't register after filling in admin_email. The log showed that I still needed to request an email address through the network.

I have no idea what to do now. What is the correct way to do it? My server has no Internet either.

**Comment by thisago** [comment]
Hello @ZimaBlueee, an login method is required, here's how I made:
```sh
# [...]
ADMIN_EMAIL=typebot@mailcatcher

SMTP_USERNAME=typebot@mailcatcher
SMTP_PASSWORD=pass
SMTP_HOST=mailcatcher
SMTP_PORT=1025
NEXT_PUBLIC_SMTP_FROM="Local Typebot' <typebot@mailcatcher>"
```
And my docker-compose:
```yml
version: "3.3"

services:
  typebot-db:
    image: postgres:14-alpine
    restart: no
    volumes:
      - ./data/postgres:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=typebot
      - POSTGRES_PASSWORD=typebot

  typebot-builder:
    image: elestio/typebot-builder:latest
    restart: no
    depends_on:
      - typebot-db
    ports:
      - "8081:3000"
    extra_hosts:
      - "host.docker.internal:host-gateway"
    env_file: .env

  typebot-viewer:
    image: elestio/typebot-viewer:latest
    restart: no
    ports:
      - "8082:3000"
    env_file: .env

  mailcatcher:
    image: dockage/mailcatcher:latest
    restart: no
    ports:
      - "1025:1025"
      - "8083:1080"
```
Then you register an Typebot account using admin email and get the code on mailcatcher page, at localhost:8083.

**Pull request merged**